Haskell/Cabal.sublime-syntax

```diff
@@ -0,0 +1,406 @@
+%YAML 1.2
+---
+# https://cabal.readthedocs.io/en/3.4/cabal-package.html
+# https://www.sublimetext.com/docs/syntax.html
+name: Cabal
+scope: source.cabal
+version: 2
+
+file_extensions:
+  - cabal
+  - cabal.project
+
+###############################################################################
+
+variables:
+  identifier: (?:[[:alpha:]][-_[:alnum:]]*)
+  module_identifier: (?:[[:upper:]][\w']*)
+
+  section_tags: |-
+    (?x:
+      common | executable | benchmark | custom-setup | flag | library | package
+    | source-repository | test-suite
+    )
+
+  boolean_tags: |-
+    (?x: benchmarks | default | optimization | tests )
+  compiler_tags: |-
+    (?x: tested-with )
+  dependency_tags: |-
+    (?x: (?:{{identifier}}-)? depends )
+  module_tags: |-
+    (?x: (?:{{identifier}}-)? modules )
+  option_tags: |-
+    (?x: (?:{{identifier}}-)? options | flags )
+  path_tags: |-
+    (?x: (?:{{identifier}}-)? (?: dir | file | path ) s? | main-is | packages )
+  url_tags: |-
+    (?x: bug-reports | homepage | location | package-url )
+  version_tags: |-
+    (?x: cabal-version | version )
+
+  compiler_values: |-
+    (?xi: GHC | UHC | LHC )
+
+  escape_chars: |-
+    (?x:
+      NUL | SOH | STX | ETX | EOT | ENQ | ACK | BEL | BS | HT | LF | VT | FF
+    | CR | SO | SI | DLE | DC1 | DC2 | DC3 | DC4 | NAK | SYN | ETB | CAN | EM
+    | SUB | ESC | FS | GS | RS | US | SP | DEL
+    | [abfnrtv\\\"'\&]
+    )
+  escape_sequence: |-
+    (?x:
+        (\\{{escape_chars}}) # Escapes
+      | (\\[0-9]+)           # Decimal Escapes
+      | (\\o[0-7]+)          # Octal Escapes
+      | (\\x[0-9A-Fa-f]+)    # Hexadecimal Escapes
+      | (\^[A-Z@\[\]\\\^_])  # Control Chars
+    )
+
+###############################################################################
+
+contexts:
+  prototype:
+    - include: comments
+
+  main:
+    - include: sections
+    # mappings
+    - include: boolean-keys
+    - include: compiler-keys
+    - include: dependency-keys
+    - include: module-keys
+    - include: option-keys
+    - include: path-keys
+    - include: url-keys
+    - include: version-keys
+    - include: other-keys
+    # expressions
+    - include: function-calls
+    - include: keywords
+    - include: operators
+
+###[ COMMENTS ]################################################################
+
+  comments:
+    - match: (--+).*\n?
```

Comment:
Maybe turn the `.*` match into a context body?

Reply:
Why?

Whether we use this simple pattern or push into a context does no longer have any performance differences. Where's the use then?

Reply:
Not consuming the entire line in a match allows for more flexibility with `with_prototype` (and inheritance nowadays). Optional in my eyes, but wanted to mention it regardless.

Reply:
That's very correct in general. I actually don't expect someone to want to inject special highlighting into comments though.

The `with_prototype` mechanism wouldn't be very useful for that anyway. An inherited syntax could easily override the whole `comments` context in case more sophisticated highlighting is desired. Otherwise `embed` is the best way incase someone want's Cabal syntax highlighting supported in e.g. Markdown fenced codeblocks.

I'd push into a context in case comments may contain special tags such as type hints in python or docstrings with dedicated tags to highlight, but that's not the case here.

Reply:
Just adding a reference for interested readers who want to read more about what FichteFoll and deathaxe are talking about:

#757 with Keith's comment there about line comments.